pkgs/applications/misc/yewtube/default.nix

```diff
@@ -0,0 +1,43 @@
+{ lib, python3Packages, fetchFromGitHub }:
+
+python3Packages.buildPythonApplication rec {
+  pname = "yewtube";
+  version = "2.9.0";
+
+  src = fetchFromGitHub {
+    owner = "iamtalhaasghar";
+    repo = "yewtube";
+    rev = "v${version}";
+    hash = "sha256-8GL2ZvRHtnnLZ07nQk3irJUj+XLL+pyUUA+JJPICPRA=";
+  };
+
+  postPatch = ''
+    substituteInPlace mps_youtube/__init__.py \
+      --replace "from pip._vendor import pkg_resources" "" \
+      --replace "__version__ =" "__version__ = '${version}' #"
+  '';
+
+  propagatedBuildInputs = with python3Packages; [
+    pyperclip
+    pyreadline
+    requests
+    youtube-search-python
+    yt-dlp
+  ];
+
+  # https://github.com/pyreadline/pyreadline/issues/74
+  doCheck = false;
+
+  # before check create a directory and redirect XDG_CONFIG_HOME to it
+  preCheck = ''
+    mkdir -p check-phase
+    export XDG_CONFIG_HOME=$(pwd)/check-phase
```

Comment:
```suggestion
    HOME=$(mktemp -d)
    export XDG_CONFIG_HOME=$HOME
```

Reply:
I changed it to `export XDG_CONFIG_HOME=$(mktemp -d)`